dali/python/nvidia/dali/pipeline.py

```diff
@@ -284,29 +284,66 @@ def outputs(self):
         self._gpu_batches_to_consume -= 1
         return self._outputs()
 
-    def _share_outputs(self):
+    def run_no_result(self):
+        """Run the pipeline without returning the resulting buffers.
+        
+        If the pipeline was created with `exec_pipelined` option set to `True`,
+        this function will also start prefetching the next iteration for
+        faster execution. It provides better control to the user about when he
+        wants to run the pipeline, when he wants to obtain resulting buffers
+        and when they can be returned to DALI pool when the results have been consumed.
+        Needs to be used together with :meth:`nvidia.dali.pipeline.Pipeline.release_outputs`
+        and :meth:`nvidia.dali.pipeline.Pipeline.share_outputs`.
+        Should not be mixed with :meth:`nvidia.dali.pipeline.Pipeline.run` in the same pipeline"""
+        if self._first_iter and self._exec_pipelined:
+            self._prefetch()
+        else:
+            self._run_once()
+
+    # for the backward compatibility
+    _run = run_no_result
+
+    def share_outputs(self):
         """Returns the outputs of the pipeline.
 
-        Main difference to outputs is that _share_outputs doesn't release
-        returned buffers, _release_outputs need to be called for that.
-        If the pipeline is executed asynchronously, this function blocks
-        until the results become available."""
+        Main difference to :meth:`nvidia.dali.pipeline.Pipeline.outputs`
```

Comment:
Maybe:
Users share the output buffers with DALI and must call :meth:`nvidia.dali.pipeline.Pipeline.release_outputs` after they finished processing them to indicate that DALI can again have exclusive access to that memory.

Reply:
Done